packages/ethereum/contracts/src/node-stake/permissioned-module/CapabilityPermissions.sol

```diff
@@ -0,0 +1,760 @@
+    // For CHANNELS target: capabilityKey (bytes32) => channel Id (keccak256(src, dest)) => GranularPermission
+    // For TOKEN target: capabilityKey (bytes32) => pair Id (keccak256(node address, spender address)) => GranularPermission
+    // For SEND target:  bytes32(0x00) => pair Id (keccak256(node address, spender address)) => GranularPermission
+    mapping(bytes32 => mapping(bytes32 => GranularPermission)) capabilities; 
+}
+
+/**
+ * @dev Drawing inspiration from the `zodiac-modifier-roles-v1` `Permissions.sol` contract, 
+ * this library is designed to support a single role and offers a set of specific functions 
+ * for interacting with HoprChannels and HoprToken contracts
+ *
+ * Adapted from `Permissions.sol` at commit 454be9d3c26f90221ca717518df002d1eca1845f, which 
+ * was audited https://github.com/gnosis/zodiac-modifier-roles-v1/tree/main/packages/evm/docs
+ *
+ * It is specifically tailored for interaction with HoprChannels and HoprToken contracts. 
+ * Additionally, it enables the transfer of native tokens to designated addresses, 
+ * while restricting the invocation of payable functions.
+ * 
+ * Some difference between this library and the original `Permissions.sol` contract are:
+ * - This library is designed to support a single role
+ * - No `DelegateCall` is allowed
+ * - Target must be one of the three types: Token, Channels, SEND
+ * - Only scoped functions are allowed. No more wildcard
+ * - Calling payable function is not allowed.
+ * - When calling HoprChannels contracts, permission is check with multiple parameters together
+ * - For Channels targets, the default permission is ALLOWED. However, the default value for other targets is BLOCKED.
+ * - Permissions are not stored bitwise in `scopeConig` (uint256) due to lack of customization
+ * - Utility functions, such as `packLeft`, `packRight`, `unpackFunction`, `unpackParameter`, `checkExecutionOptions` are removed
+ * - Specific helper functions, such as `pluckOneStaticAddress`, `pluckTwoStaticAddresses`, `pluckDynamicAddresses`,  `pluckSendPayload` are derived from `pluckStaticValue` and `pluckDynamicValue`
+ * - helper functions to encode array of function signatures and their respective permissions are added.
+ *
+ * @notice Due to the deployed HoprToken.sol imports OpenZeppelin contract library locked at v4.4.2, while
+ * HoprChannels contract imports OpenZeppelin contract of v4.8.3, it's not possible to import both contracts
+ * the same time without creating conflicts. Therefore, two method identifiers of HoprToken contract are
+ * defined with value instead of `.selector`
+ */
+library HoprCapabilityPermissions {
+    using TargetUtils for Target;
+    using EnumerableTargetSet for TargetSet;
+
+    // HoprChannels method ids (TargetType.CHANNELS)
+    bytes4 internal constant REDEEM_TICKET_SELECTOR = HoprChannels.redeemTicketSafe.selector;
```

Comment:
Why don't use `HoprChannels.redeemTicketSafe.selector` directly and let the compiler do the "caching". This creates unnecessary extra code

Reply:
This is for clarity only as the selector is used multiple times in the code